src/time_steppers/backwardEuler.cpp

```diff
@@ -94,67 +95,92 @@ void backwardEuler::lineSearch(double dt) {
     }
     // Initialize an interval for optimal learning rate alpha
     double amax = 2;
-    double amin = 1e-3;
-//    double amin = 1e-5;
+    double amin = 1e-3;    
     double al = 0;
     double au = 1;
-
     double a = 1;
 
     //compute the slope initially
     double q0 = 0.5 * pow(Force.norm(), 2);
     double dq0 = -(Force.transpose() * Jacobian * DX)(0);
 
-    bool success = false;
-    double m2 = 0.9;
-    double m1 = 0.1;
-    int iter_l = 0;
-    while (!success) {
-        int limb_idx = 0;
-        for (auto& joint : joints) {
-            joint->x = joint->x_ls;
-        }
-        for (auto& limb : limbs) {
-            limb->x= limb->x_ls;
-            limb->updateNewtonX(dx, offsets[limb_idx], alpha);
```

Comment:
Bug 2: The previous code base incorrectly used `alpha` here instead of `a`!!! This bug has gone undetected for a very long time.